### PR TITLE
SCPUI needs to be able run game_close_level

### LINKED ISSUE #4755: Mainhalls stop animating under particular SCPUI condition
Using SCPUI for any mission-loaded UI and then backing out of the mission causes the mainhall animations to stop animating until a mission is played and completed.

Specific steps to reproduce. 

1) Utilize the SCPUI mod found [here](https://www.hard-light.net/forums/index.php?topic=98473.0) and launch on a very recent nightly. It should launch out of the box into retail + scpui
2) Start a mission by clicking on the ready room. Any mission, doesn't matter.
3) Back out of the mission by pressing Esc
4) Mouseover any animated mainhall element. They will not animate, but the sound will be played and clicking still works.

I have tried to track this one down on and off for a month and have been unsuccessful. My guess is it has something to do with a librocket context not getting completely unloaded.

EDIT: This issue is a case-study in how keywords cause SCP to ignore tickets.

### PR DESCRIPTION
**Pull request by MjnMixael**
Fixes #4755 ~~by making sure the timer is not ever paused during a mainhall game state. We can't rely on UI-specific code to restart the timer after mission load if SCPUI is being used, so it should be done whenever the mainhall is entered probably. I didn't do a deep dive into the timer functions, so maybe there's a more appropriate fix elsewhere.~~

-by adding the ability to run `game_level_close()` from the scripting interface.

### CODE REVIEW

**Comment by Goober5000** [comment]
I'll take a look at this.  The timer code can be dicey, so I want to look at some of the possible code flow paths.

**Comment by Goober5000** [comment]
#4755 is going to need a more comprehensive fix.  I've posted a comment there.

**Comment by MjnMixael** [comment]
> #4755 is going to need a more comprehensive fix. I've posted a comment there.

Not that comprehensive. Just give SCPUI the ability to call the level close method. Tested and works.

**Comment by Goober5000** [comment]
Hang on... I ran a search just now for `game_level_close()` and there is already a `mn.unloadMission()` function.  I missed this earlier because I searched for `freespace_stop_mission()`.

**Comment by MjnMixael** [comment]
> mn.unloadMission()

That method does not work presumably because SCPUI doesn't pass the `if(Game_mode & GM_IN_MISSION)` test

**Comment by MjnMixael** [comment]
can't wait to see why this fix won't work either

File: code/scripting/api/libs/mission.cpp [hunk 1]
```diff
@@ -1464,11 +1464,13 @@ ADE_FUNC(loadMission, l_Mission, "string missionName", "Loads a mission", "boole
 	return ADE_RETURN_TRUE;
 }
 
-ADE_FUNC(unloadMission, l_Mission, NULL, "Stops the current mission and unloads it", NULL, NULL)
+ADE_FUNC(unloadMission, l_Mission, "boolean forceUnload", "Stops the current mission and unloads it. If forceUnload is true "
```
Comment (Goober5000) [inline code/scripting/api/libs/mission.cpp:1467]:
```suggestion
ADE_FUNC(unloadMission, l_Mission, "[boolean forceUnload]", "Stops the current mission and unloads it. If forceUnload is true "
```

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Looks good.  Only thing is that the documentation should reflect that the parameter is optional.

File: code/scripting/api/libs/mission.cpp (hunk 1 above)
Comment (MjnMixael) [reply code/scripting/api/libs/mission.cpp]: Done

**Review by Goober5000 — APPROVED** [review]

**Review by JohnAFernandez — APPROVED** [review]

**Pull request merged**